### PR TITLE
Simulation and passive DNF plugin

### PR DESCRIPTION
**Pull request by onionka**

### CODE REVIEW

**Comment by onionka** [comment]
Here is [plot](https://github.com/rh-lab-q/deltametadata-prototype/files/267834/plot.pdf) of how much data in bytes it has to download, when synchronizing from 20.4.2016 to `x-axis`. But we've got very little sample of repodata so it is inaccurate.

**Comment by mluscon** [comment]
Please fix a corner case in the plugin - if destination file does not exists, zsync will not create it.

**Comment by onionka** [comment]
OK it is done, and now if something is not synchronized well, it will just print report to stderr and continues

File: client/dnf_zsync.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            return -1
+        try:
+            zsync = Popen(["zsync", url, "-i", input_file, "-o",
+                           target], stdout=PIPE, stderr=PIPE)
+            outputs = zsync.communicate()
+            if self._print_log:
+                print(outputs[1].decode("utf-8"))
+                print(outputs[0].decode("utf-8"))
+        except CalledProcessError as ex:
+            print(str(ex), file=sys.stderr)
```
Comment (mluscon) [inline client/dnf_zsync.py]: We need to make the plugin compatible with python2 too - there is no file attribute of print in python2.

File: client/dnf_zsync.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
+import os.path
+import re
+import sys
+from subprocess import DEVNULL, PIPE, CalledProcessError, Popen, check_output
```
Comment (mluscon) [inline client/dnf_zsync.py]: DEVNULL is also not available in python2.

**Comment by mluscon** [comment]
> OK it is done, and now if something is not synchronized well, it will just print report to stderr and continues

I am sorry for the misleading comment. I meant not to print an error but correctly download the missing file. Just from the top of my head, you can create a zero length file before calling zsync if the file does not exist.

**Comment by mluscon** [comment]
merged

**Pull request closed without merging**